src/schoology-extractor/schoology_extractor/mapping/discussions.py

```diff
@@ -0,0 +1,88 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import pandas as pd
+
+from . import constants
+
+DISCUSSION_TYPE = "Discussion"
+
+
+def map_to_udm(discussions_df: pd.DataFrame, section_id: int) -> pd.DataFrame:
+    """
+    Maps a DataFrame containing Schoology Discussion threads
+    into the Ed-Fi LMS Unified Data Model (UDM) format.
+
+    Parameters
+    ----------
+    discussions_df: DataFrame
+        Pandas DataFrame containing Schoology discussions
+
+    Returns
+    -------
+    DataFrame
+        A LMSUsers-formatted DataFrame
+
+    Notes
+    -----
+    DataFrame columns are:
+        SourceSystemIdentifier: A unique number or alphanumeric code assigned to a the discussion-reply by
+            the source system
+        SourceSystem: The system code or name providing the user data
+        LMSUserIdentifier: A unique number or alphanumeric code assigned to a user by the source
+            system
+        LMSSectionIdentifier: A unique number or alphanumeric code assigned to a section by the
+            source system
+        EntityStatus: The status of the record
+        ActivityDateTime: The date/time the replied was created.
+        ActivityStatus: The status for the reply
+        ActivityType: The type of activity: `Discussion reply`
+        Content: The comment text.
+        AssignmentIdentifier: A unique numeric identifier assigned to the assignment.
+        ActivityTimeInMinutes: The total activity time in minutes.
+        CreateDate: Date/time at which the record was first retrieved
+        LastModifieDate: Date/time when the record was modified, or when first retrieved
+    """
+    df = discussions_df[[
+        "completed",
+        "graded",
+        "available",
+        "published",
+        "uid",
+        "id",
+        "CreateDate",
+        "LastModifiedDate"
+        ]].copy()
+
+    df["created"] = df["CreateDate"]
+    df["status"] = None
+    df["status"] = df["published"].apply(lambda x: 'published' if x == 1 else df["status"])
+    df["status"] = df["available"].apply(lambda x: 'available' if x == 1 else df["status"])
+    df["status"] = df["graded"].apply(lambda x: 'graded' if x == 1 else df["status"])
+    df["status"] = df["completed"].apply(lambda x: 'completed' if x == 1 else df["status"])
```

Comment:
It might be more efficient to write a function that takes a row as input and returns a string

```python
def _get_status(row: pd.Series) -> str:
    # access row["status"], row["published"], etc.

df["status] = df.apply(_get_status, axis=1)
```

Reply:
✔️